src/Aspire.Hosting/Dashboard/ConsoleLogPublisher.cs

```diff
@@ -14,13 +25,38 @@ internal IAsyncEnumerable<IReadOnlyList<(string Content, bool IsErrorMessage)>>?
         }
 
         // Obtain logs using the relevant approach.
-        // Note, we would like to obtain these logs via DCP directly, rather than sourcing them in the dashboard.
-        return resource switch
+        if (UseLogStreaming)
+        {
+            return resource switch
+            {
+                ExecutableSnapshot executable => SubscribeExecutableResource(executable),
+                ContainerSnapshot container => SubscribeContainerResource(container),
+                _ => throw new NotSupportedException($"Unsupported resource type {resource.GetType()}.")
+            };
+        }
+        else
+        {
+#pragma warning disable CS0162 // Unreachable code detected
+            return resource switch
+            {
+                ExecutableSnapshot executable => SubscribeExecutable(executable),
+                ContainerSnapshot container => SubscribeContainer(container),
+                _ => throw new NotSupportedException($"Unsupported resource type {resource.GetType()}.")
+            };
+#pragma warning restore CS0162 // Unreachable code detected
+        }
```

Comment:
What's going on here with UseLogStreaming?

Reply:
A suggestion from @davidfowl , meant to facilitate quick switch back to old ways of getting the logs from Containers and Executables in case we detect issues with the new approach late in the game.